enhancements/storage/storage-hypershift.md

```diff
@@ -0,0 +1,973 @@
+| awso/assets/rbac/privileged_role.yaml                                 | AWSO                   | G         | RBAC for the CSI driver DaemonSet to be privileged                                                                                                                  |
+| awso/assets/rbac/prometheus_role.yaml                                 | AWSO? (not in MVP)     | MC / N/A? | RBAC for Prometheus to scan CSI driver namespace.                                                                                                                   |
+| awso/assets/rbac/prometheus_rolebinding.yaml                          | AWSO? (not in MVP)     | MC / N/A? | RBAC for Prometheus to scan CSI driver namespace.                                                                                                                   |
+| awso/assets/rbac/provisioner_binding.yaml                             | AWSO? (not in MVP)     | G?        | RBAC for the CSI driver Deployment in the management cluster to watch/get/update/create objects in the guest cluster                                                |
+| awso/assets/rbac/provisioner_role.yaml                                | AWSO? (not in MVP)     | G?        | RBAC for the CSI driver Deployment in the management cluster to watch/get/update/create objects in the guest cluster                                                |
+| awso/assets/rbac/resizer_binding.yaml                                 | AWSO? (not in MVP)     | G?        | RBAC for the CSI driver Deployment in the management cluster to watch/get/update/create objects in the guest cluster                                                |
+| awso/assets/rbac/resizer_role.yaml                                    | AWSO? (not in MVP)     | G?        | RBAC for the CSI driver Deployment in the management cluster to watch/get/update/create objects in the guest cluster                                                |
+| awso/assets/rbac/snapshotter_binding.yaml                             | AWSO? (not in MVP)     | G?        | RBAC for the CSI driver Deployment in the management cluster to watch/get/update/create objects in the guest cluster                                                |
+| awso/assets/rbac/snapshotter_role.yaml                                | AWSO? (not in MVP)     | G?        | RBAC for the CSI driver Deployment in the management cluster to watch/get/update/create objects in the guest cluster                                                |
+| NEW: AWS cloud credentials                                            | CPO                    | MC        | AWS credentials for the operator + driver                                                                                                                           |
```

Comment:
These get laid down by the HCCO and should be no change from what happens today, no? We currently create the credentials secret with the role arn for storage.

Reply:
Do you create them in the management cluster? If so, then yes, it should not be needed.
I thought HCCO creates them in the guest cluster. Then we need to remove this, because we don't need any AWS credentials there.

Reply:
The credentials don't need to live in the management cluster for you to use them, and I guess it comes down to your specific operator. For example, we run the ingress operator on the management cluster, but create the credentials secret inside the guest cluster. Because the ingress operator knows where to find its credentials it reads them from the guest cluster and uses them. What changes is how the webidentity token is generated. In the standalone case, the ingress operator simply includes a projected volume with service account token. In the hpershift case, we attach a token-minter sidecar to the operator deployment.

Reply:
We need the credentials in the control-plane Deployment of AWS EBS CSI driver, currently targeting the management cluster. We can add a minter sidecar to it. But it still needs a Secret with:

```
    [default]
    role_name: <driver-role-name> 
    web_identity_token_file: <path-to-a-minted>
```

And something needs to create the actual `<driver-role-name>` in AWS. It must be already solved problem in control-plane-operator, I don't expect any issue here, just the Secret must be created by CPO.